app/models/storage.rb

```diff
@@ -70,6 +71,12 @@ class Storage < ApplicationRecord
 
   SUPPORTED_STORAGE_TYPES = %w( VMFS NFS FCP ISCSI GLUSTERFS )
 
+  supports :smartstate_analysis do
+    if ext_management_systems.blank? || !ext_management_system.kind_of?(ManageIQ::Providers::Vmware::InfraManager)
```

Comment:
There is a VMware Datastore Browser API but I think we basically ssh to the host and run `ls -l` :smile:
@roliveri should be able to explain better